WordPress/Classes/ViewRelated/Post/PostSharingController.swift

```diff
@@ -145,3 +145,26 @@ import SVProgressHUD
 
     typealias PopoverAnchor = UIPopoverPresentationController.PopoverAnchor
 }
+
+private class CustomActivityItemProvider: UIActivityItemProvider {
+    private let title: String?
+    private let summary: String?
+    private let url: URL?
+
+    init(title: String?, summary: String?, url: URL?) {
+        self.title = title
+        self.summary = summary
+        self.url = url
+        super.init(placeholderItem: url ?? "")
+    }
+
+    override var item: Any {
+        guard let activityType = self.activityType else { return url ?? "" }
+
+        if activityType == .copyToPasteboard {
+            return url ?? ""
+        } else {
+            return [title as Any, summary as Any, url as Any].compactMap { $0 }
```

Comment:
This looks very ugly but the compiler is firing a warning here due to strings being optional and I had to silence the warning one way or the other. LMK if you prefer another way to go about it